src/index.ts

```diff
@@ -66,6 +66,19 @@ export class Collection<K, V> extends Map<K, V> {
 		return super.clear();
 	}
 
+	/**
+	 * Gets an element if the key exists, otherwise sets and returns {@param defaultValue}.
+	 * @param {*} key - Key to get from/set to the collection.
+	 * @param {*} defaultValue - Default value to be set and returned if the key doesn't exist.
+	 * @returns {*} The existing value if any, {@param defaultValue} otherwise.
+	 */
+	public ensure(key: K, defaultValue: V): V {
```

Comment:
Consider either making this `() => V` or another function with that instead for lazily evaluating the default.

Reply:
Your change does not solve this. `defaultValue` is evaluated at the function call, not when used. It should look like `coll.ensure(k, () => v)`.

Reply:
My bad, misread default as result.